ci/infra/testrunner/skuba/skuba.py

```diff
@@ -132,8 +138,48 @@ def _wait_node_joined(self, role, node, timeout=60, backoff=10):
                 raise Exception((f'Node {node_name} not shown ready after {timeout} seconds'
                                  f'{". Last error:"+str(last_error) if last_error else ""}'))
             time.sleep(backoff)
+
+
+    def _wait_apiserver_ready(self, node, timeout=60, backoff=10):
+        apiserver_healthz = 'curl -Ls --insecure https://localhost:6443/healthz'
+
+        try:
+            self._wait_condition("master", node, apiserver_healthz, 'ok', timeout=timeout)
+        except AssertionError as ex:
+            node_name = self.platform.get_nodes_names("master")[node]
+            raise Exception(f'Error waiting apiserver at {node_name} to become ready: {ex}')
+
+
+    def _wait_etcd_ready(self, node, timeout=60, backoff=10):
+        etcd_health_check = ('sudo curl -Ls --cacert /etc/kubernetes/pki/etcd/ca.crt '
+                             '--key /etc/kubernetes/pki/etcd/server.key '
+                             '--cert /etc/kubernetes/pki/etcd/server.crt '
+                             'https://localhost:2379/health')
+
+        try:
+           self._wait_condition("master", node, etcd_health_check, 'true', timeout=timeout)
+        except AssertionError as ex:
+            node_name = self.platform.get_nodes_names("master")[node]
+            raise Exception(f'Error waiting etcd at {node_name} to become ready: {ex}')
         
+
+    def _wait_condition(self, role, node, command, condition, timeout=60, backoff=10):
```

Comment:
should we just make a decorator so we can apply it to needed functions?

Reply:
It could be but I would postpose that to another PR.

Reply:
Agreed, was not meant as in this PR but as a public question :)